### ISSUE TITLE
Feature Request: Consolidate tf.keras get_updates_for with existing tensorflow update_ops

### ISSUE THREAD

**Issue by ankitvgupta** [tensorflow/tensorflow#24050]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): TF 1.12
- Are you willing to contribute it (Yes/No): Potentially (not sure how complicated this is)

**Describe the feature and the current behavior/state.**
Currently, if `tf.keras` layers have update_ops (notably BatchNormalization), they are not added to `tf.GraphKeys.UPDATE_OPS`. Instead, we have to directly access them via `tf.keras.Model.get_updates_for`. However, for normal tf.layers implementations like https://www.tensorflow.org/api_docs/python/tf/layers/batch_normalization, these ops are directly added to `tf.GraphKeys.UPDATE_OPS` and this is cleanly documented. 

Is there a reason this can't be done with `tf.keras` models too? Or at least be much better documented?

**Will this change the current api? How?**
No, we can keep the current API. But, this will allow users migrating from older implementations to tf.keras based models to maintain their update_ops control dependencies.

**Who will benefit with this feature?**
Developers will face less technical load in migrating to `tf.keras`, especially since this was not clearly documented in the `tf.keras.layers.BatchNormalization` page.

**Any Other info.**

**Comment by ankitvgupta** [comment]
After looking at more docs, it looks like this was an intentional design choice. Can this, at least, be better documented in the tf.keras docs? Especially since there are now examples of using tf.keras layers in custom estimators, where these update ops need to be handled.

**Comment by omalleyt12** [comment]
The global collections are going away in TF2.0, so I'm not sure what we could do here. Agree better doc would be good
